### PR TITLE
Use `externalMailInfo` API call instead of `mailboxes` one to get ExternalMail info

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #1669

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageExternalsManager.kt [hunk 1]
```diff
@@ -50,27 +50,40 @@ class NewMessageExternalsManager @Inject constructor() : NewMessageManager() {
...
+            bccField.updateExternals(shouldWarnForExternal, emailDictionary, aliases, trustedDomains)
         }
     }
 
-    private fun updateBanner(shouldWarnForExternal: Boolean, emailDictionary: MergedContactDictionary, aliases: List<String>) {
+    private fun updateBanner(
+        shouldWarnForExternal: Boolean,
+        emailDictionary: MergedContactDictionary,
+        aliases: List<String>,
+        trustedDomains: List<String>,
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageExternalsManager.kt:80]: Maybe we could make a dataclass out of these 4 parameters to simplify the 3 functions declarations

**Review by FabianDevel — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageExternalsManager.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageExternalsManager.kt]: Finally, out of the last 3. It was better overall.

File: app/src/main/java/com/infomaniak/mail/data/models/correspondent/Recipient.kt [hunk 2]
```diff
@@ -64,13 +64,13 @@ open class Recipient : EmbeddedRealmObject, Correspondent {
 
     // Computes if the Recipient is external, according to the required conditions.
     // Does not tell anything about how to display the Recipient chip when composing a new Message.
-    fun isExternal(emailDictionary: MergedContactDictionary, aliases: List<String>, trustedDomains: List<String>): Boolean {
+    fun isExternal(externalData: ExternalData): Boolean = with(externalData) {
         val isUnknownContact = email !in emailDictionary
-        val isMailerDaemon = """mailer-daemon@(?:.+\.)?infomaniak\.ch""".toRegex(RegexOption.IGNORE_CASE).matches(email)
-        val isUntrustedDomain = email.isEmail() && trustedDomains.none { email.endsWith(it) }
         val isAlias = email in aliases
+        val isUntrustedDomain = email.isEmail() && trustedDomains.none { email.endsWith(it) }
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/data/models/correspondent/Recipient.kt:70]: Use reference

File: app/src/main/java/com/infomaniak/mail/utils/ExternalUtils.kt [hunk 3]
```diff
@@ -49,15 +43,11 @@ object ExternalUtils {
...
-        aliases: List<String>,
-        emailDictionary: MergedContactDictionary,
-        trustedDomains: List<String>,
-    ): Pair<String?, Int> {
+    fun Draft.findExternalRecipientForNewMessage(externalData: ExternalData): Pair<String?, Int> {
         val to = to.onlyAutomaticallyAddedOnes()
         val cc = cc.onlyAutomaticallyAddedOnes()
         val bcc = bcc.onlyAutomaticallyAddedOnes()
-        return findExternalRecipientInIterables(emailDictionary, aliases, trustedDomains, to, cc, bcc)
+        return findExternalRecipientInIterables(externalData, to, cc, bcc)
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/utils/ExternalUtils.kt:50]: Jump line

**Review by FabianDevel — CHANGES_REQUESTED** [review]

**Review by FabianDevel — APPROVED** [review]

**Pull request merged**